src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -85,8 +85,19 @@ static String clusterNameFromDDLQuery(ContextPtr context, const DDLTask & task)
 
     String description = fmt::format("from {}", task.entry_path);
     ParserQuery parser_query(end, settings[Setting::allow_settings_after_format_in_insert]);
-    ASTPtr query = parseQuery(
-        parser_query, begin, end, description, settings[Setting::max_query_size], settings[Setting::max_parser_depth], settings[Setting::max_parser_backtracks]);
+    ASTPtr query;
+
+    try
+    {
+        query = parseQuery(
+            parser_query, begin, end, description, settings[Setting::max_query_size], settings[Setting::max_parser_depth], settings[Setting::max_parser_backtracks]);
+    }
+    catch (Exception &)
```

Comment:
```suggestion
    catch (const Exception &)
```

Also shouldn't we ignore only `SYNTAX_ERROR`?

Reply:
==
ilejn@ilejnThinkPad:~/projects/ClickHouse/src$ grep -R 'catch (const Exception' . --include \*cpp | wc -l
144
ilejn@ilejnThinkPad:~/projects/ClickHouse/src$ grep -R 'catch (Exception' . --include \*cpp | wc -l
209
==

I'll add const of course.
Just for fun.

Regarding restricting by SYNTAX_ERROR. 
To be honest I cannot imagine why/how it can be useful.

Reply:
Catching exceptions too broadly already shows problems in other places, here I guess it is not that critical, but still. For instance imagine it is `LOGICAL_ERROR` somehow, we definitely should not catch it.

Reply:
> Catching exceptions too broadly already shows problems in other places, here I guess it is not that critical, but still. For instance imagine it is `LOGICAL_ERROR` somehow, we definitely should not ignore it.

Agree.
We can imagine that a query from an old version causes a logical error in a modern version.
So this change supposed to switch from showing nothing to showing something, not to crashing.

Reply:
>We can imagine that a query from an old version causes a logical error in a modern version.

This should not be possible, but it can be true
For instance this change also ignores the `MEMORY_LIMIT_EXCEEDED` error which is not desired behavior I would say
Also it ignores errors on exceeding `max_parse_depth` and `max_parser_backtracks` which is also bad, if it fails for this reason, user should be notified, so that settings could be increased

So TL;DR; - let's ignore only `SYNTAX_ERROR` for now, I think it is OK (and it also will fix the issue for you AFAICS)

Reply:
To me ignoring any error including MEMORY_LIMIT_EXCEEDED is exactly desired behavior:
we cannot find 'cluster' out and the best what we can do is to show query as is.

Anyway, ok, lets catch syntax error only.